language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -111,19 +244,27 @@ lookupDynamicBuiltinName dynName = do
 -- 2. calls 'returnCek' on values ('TyAbs', 'LamAbs', 'Constant')
 -- 3. returns 'EvaluationFailure' ('Error')
 -- 4. looks up a variable in the environment and calls 'returnCek' ('Var')
-computeCek :: Context -> Plain Term -> CekM EvaluationResultDef
-computeCek con (TyInst _ body ty)       = computeCek (FrameTyInstArg ty : con) body
-computeCek con (Apply _ fun arg)        = do
+computeCek :: Context -> WithMemory Term -> CekM w s EvaluationResultDef
+computeCek con t@(TyInst _ body ty)       = do
+    spendBoth t 1 1
```

Comment:
Do we want to say that reduction steps uses memory? I'd also make named constants for all constant values, so it's easy to tweak, e.g. `reductionStepCPU`.